astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -194,6 +194,116 @@ def test_cirs_to_hadec():
     assert_allclose(cirs.dec, cirs3.dec)
 
 
+def test_itrs_topo_to_altaz_with_refraction():
+
+    loc = EarthLocation(lat=0*u.deg, lon=0*u.deg, height=0*u.m)
```

Comment:
Might it be worth to use a few more locations? To keep total time the same, maybe reduce golden spiral to 40 and here do
```
EarthLocation(lat=[[0], [90], [335], [-60], [-75]] * u.deg,
              lon=[[0], [0], [45], [89], [-60]] * u.deg,
              height=[[0], [1], [-0.1], [0.4], [2]] * u.km)
```

Reply:
Sure, but I am curious as to why you feel these tests would benefit from the use of multiple locations when many other tests just use one?

Reply:
Also, using a latitude of 335 degrees is not going to work. I'm going to change that to 33.5.

Reply:
Sorry for the poor example! I always give longitude first and clearly didn't even look at the name... 

More on topic: I think it is good to execute the array option for parameters, mostly *because* most tests do not do it! But obviously there really was nothing wrong with the test as you had it!

Reply:
Not working!
```
E                   ValueError: attribute location should be scalar or have shape (40,), but is has shape (5, 1) and could not be broadcast.
```

Reply:
@mhvk I could use your help with this.

Reply:
@mkbrewer - I think best may be just to revert to what you had -- any problem here is certainly not related to your PR. But would be great if you could raise a separate iisue about this shape incompatibility problem (am very much on and off in buses myself, sorry!)

Reply:
Actually, I think my suggestion was just wrong, and there is already an issue about changing the behaviour that attributes like `location` have to be broadcastable to the data. Sorry for this!!

Reply:
Just in case anyone wants to follow this up, see #8812

Reply:
OK. I will revert then. Here are the two errors, for future reference: 
```
2022-07-26T13:38:41.0002909Z ___________________ test_itrs_topo_to_altaz_with_refraction ____________________
2022-07-26T13:38:41.0003121Z 
2022-07-26T13:38:41.0003279Z     def test_itrs_topo_to_altaz_with_refraction():
2022-07-26T13:38:41.0003711Z     
2022-07-26T13:38:41.0004333Z         locs = EarthLocation(lat=[[0], [90], [33.5], [-60], [-75]] * u.deg,
2022-07-26T13:38:41.0004776Z                              lon=[[0], [0], [45], [89], [-60]] * u.deg,
2022-07-26T13:38:41.0005196Z                              height=[[0], [1], [-0.1], [0.4], [2]] * u.km)
2022-07-26T13:38:41.0005485Z         usph = golden_spiral_grid(40)
2022-07-26T13:38:41.0005803Z         dist = np.linspace(1., 1000.0, len(usph)) * u.au
2022-07-26T13:38:41.0006157Z         icrs = ICRS(ra=usph.lon, dec=usph.lat, distance=dist)
2022-07-26T13:38:41.0006603Z         altaz_frame1 = AltAz(obstime = 'J2000', location=locs)
2022-07-26T13:38:41.0007106Z         altaz_frame2 = AltAz(obstime = 'J2000', location=locs, pressure=1000.0 * u.hPa,
2022-07-26T13:38:41.0007722Z                              relative_humidity=0.5)
2022-07-26T13:38:41.0008133Z         cirs_frame = CIRS(obstime = 'J2000', location=locs)
2022-07-26T13:38:41.0008432Z         itrs_frame = ITRS(location=locs)
2022-07-26T13:38:41.0008678Z     
2022-07-26T13:38:41.0008894Z         # Normal route
2022-07-26T13:38:41.0009121Z         # No Refraction
2022-07-26T13:38:41.0009407Z         altaz1 = icrs.transform_to(altaz_frame1)
2022-07-26T13:38:41.0009672Z     
2022-07-26T13:38:41.0009879Z         # Refraction added
2022-07-26T13:38:41.0010170Z         altaz2 = icrs.transform_to(altaz_frame2)
2022-07-26T13:38:41.0010429Z     
2022-07-26T13:38:41.0010638Z         # Refraction removed
2022-07-26T13:38:41.0010931Z >       cirs = altaz2.transform_to(cirs_frame)
2022-07-26T13:38:41.0011117Z 
2022-07-26T13:38:41.0011570Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/tests/test_intermediate_transformations.py:219: 
2022-07-26T13:38:41.0012027Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0012573Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/baseframe.py:1202: in transform_to
2022-07-26T13:38:41.0012971Z     return trans(self, new_frame)
2022-07-26T13:38:41.0013523Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/transformations.py:1478: in __call__
2022-07-26T13:38:41.0013931Z     curr_coord = t(curr_coord, curr_toframe)
2022-07-26T13:38:41.0014496Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/transformations.py:1079: in __call__
2022-07-26T13:38:41.0014901Z     return supcall(fromcoord, toframe)
2022-07-26T13:38:41.0015542Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/builtin_frames/cirs_observed_transforms.py:76: in observed_to_cirs
2022-07-26T13:38:41.0016107Z     astrom = erfa_astrom.get().apio(observed_coo)
2022-07-26T13:38:41.0016425Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0016605Z 
2022-07-26T13:38:41.0016813Z frame_or_coord = <AltAz Coordinate (obstime=J2000.000, location=[[(6.37813700e+03,     0.        ,     0.        )
2022-07-26T13:38:41.0017148Z   (6.37813700e+03,  ...8762211e+02),
2022-07-26T13:38:41.0017398Z       (104.26257895,  69.02798708, 9.75082339e+02),
2022-07-26T13:38:41.0017670Z       (228.70516507,  73.75610056, 1.00023686e+03)]]>
2022-07-26T13:38:41.0017832Z 
2022-07-26T13:38:41.0017927Z     @staticmethod
2022-07-26T13:38:41.0018160Z     def apio(frame_or_coord):
2022-07-26T13:38:41.0018540Z         '''
2022-07-26T13:38:41.0018995Z         Slightly modified equivalent of ``erfa.apio``, used in conversions AltAz <-> CIRS.
2022-07-26T13:38:41.0019350Z     
2022-07-26T13:38:41.0019686Z         Since we use a topocentric CIRS frame, we have dropped the steps needed to calculate
2022-07-26T13:38:41.0020044Z         diurnal aberration.
2022-07-26T13:38:41.0020262Z     
2022-07-26T13:38:41.0020477Z         Parameters
2022-07-26T13:38:41.0020757Z         ----------
2022-07-26T13:38:41.0021171Z         frame_or_coord : ``astropy.coordinates.BaseCoordinateFrame`` or ``astropy.coordinates.SkyCoord``
2022-07-26T13:38:41.0021664Z             Frame or coordinate instance in the corresponding frame
2022-07-26T13:38:41.0022048Z             for which to calculate the calculate the astrom values.
2022-07-26T13:38:41.0022409Z             For this function, an AltAz frame is expected.
2022-07-26T13:38:41.0022706Z         '''
2022-07-26T13:38:41.0022985Z         # Calculate erfa.apio input parameters.
2022-07-26T13:38:41.0023271Z         # TIO locator s'
2022-07-26T13:38:41.0023660Z         sp = erfa.sp00(*get_jd12(frame_or_coord.obstime, 'tt'))
2022-07-26T13:38:41.0023941Z     
2022-07-26T13:38:41.0024171Z         # Earth rotation angle.
2022-07-26T13:38:41.0024585Z         theta = erfa.era00(*get_jd12(frame_or_coord.obstime, 'ut1'))
2022-07-26T13:38:41.0024873Z     
2022-07-26T13:38:41.0025123Z         # Longitude and latitude in radians.
2022-07-26T13:38:41.0025651Z         lon, lat, height = frame_or_coord.location.to_geodetic('WGS84')
2022-07-26T13:38:41.0025989Z         elong = lon.to_value(u.radian)
2022-07-26T13:38:41.0026275Z         phi = lat.to_value(u.radian)
2022-07-26T13:38:41.0026514Z     
2022-07-26T13:38:41.0026764Z         # Polar motion, rotated onto local meridian
2022-07-26T13:38:41.0027108Z         xp, yp = get_polar_motion(frame_or_coord.obstime)
2022-07-26T13:38:41.0027377Z     
2022-07-26T13:38:41.0027670Z         # we need an empty astrom structure before we fill in the required sections
2022-07-26T13:38:41.0028110Z         astrom = np.zeros(frame_or_coord.obstime.shape, dtype=erfa.dt_eraASTROM)
2022-07-26T13:38:41.0028424Z     
2022-07-26T13:38:41.0028696Z         # Form the rotation matrix, CIRS to apparent [HA,Dec].
2022-07-26T13:38:41.0029131Z         r = (rotation_matrix(elong, 'z', unit=u.radian)
2022-07-26T13:38:41.0029546Z              @ rotation_matrix(-yp, 'x', unit=u.radian)
2022-07-26T13:38:41.0029940Z              @ rotation_matrix(-xp, 'y', unit=u.radian)
2022-07-26T13:38:41.0030361Z              @ rotation_matrix(theta+sp, 'z', unit=u.radian))
2022-07-26T13:38:41.0030629Z     
2022-07-26T13:38:41.0030921Z         # Solve for local Earth rotation angle.
2022-07-26T13:38:41.0031180Z         a = r[..., 0, 0]
2022-07-26T13:38:41.0031420Z         b = r[..., 0, 1]
2022-07-26T13:38:41.0031671Z         eral = np.arctan2(b, a)
2022-07-26T13:38:41.0031980Z >       astrom['eral'] = eral
2022-07-26T13:38:41.0032438Z E       ValueError: could not broadcast input array from shape (5,40) into shape ()
```
And:
```
2022-07-26T13:38:41.0195668Z _____________________________ test_cirs_itrs_topo ______________________________
2022-07-26T13:38:41.0195818Z 
2022-07-26T13:38:41.0196036Z self = <astropy.coordinates.attributes.EarthLocationAttribute object at 0x7ff05c885390>
2022-07-26T13:38:41.0196671Z instance = <[ValueError('attribute location should be scalar or have shape (40,), but is has shape (5, 1) and could not be broadcast.') raised in repr()] CIRS object at 0x7ff0502b52a0>
2022-07-26T13:38:41.0197166Z frame_cls = <class 'astropy.coordinates.builtin_frames.cirs.CIRS'>
2022-07-26T13:38:41.0197355Z 
2022-07-26T13:38:41.0197468Z     def __get__(self, instance, frame_cls=None):
2022-07-26T13:38:41.0197691Z         if instance is None:
2022-07-26T13:38:41.0197881Z             out = self.default
2022-07-26T13:38:41.0198062Z         else:
2022-07-26T13:38:41.0198359Z             out = getattr(instance, '_' + self.name, self.default)
2022-07-26T13:38:41.0198578Z             if out is None:
2022-07-26T13:38:41.0198834Z                 out = getattr(instance, self.secondary_attribute, self.default)
2022-07-26T13:38:41.0199237Z     
2022-07-26T13:38:41.0199535Z         out, converted = self.convert_input(out)
2022-07-26T13:38:41.0199761Z         if instance is not None:
2022-07-26T13:38:41.0200307Z             instance_shape = getattr(instance, 'shape', None)  # None if instance (frame) has no data!
2022-07-26T13:38:41.0200799Z             if instance_shape is not None and (getattr(out, 'shape', ()) and
2022-07-26T13:38:41.0201078Z                                                out.shape != instance_shape):
2022-07-26T13:38:41.0201353Z                 # If the shapes do not match, try broadcasting.
2022-07-26T13:38:41.0201577Z                 try:
2022-07-26T13:38:41.0201798Z                     if isinstance(out, ShapedLikeNDArray):
2022-07-26T13:38:41.0202092Z                         out = out._apply(np.broadcast_to, shape=instance_shape,
2022-07-26T13:38:41.0202353Z                                          subok=True)
2022-07-26T13:38:41.0202537Z                     else:
2022-07-26T13:38:41.0202782Z >                       out = np.broadcast_to(out, instance_shape, subok=True)
2022-07-26T13:38:41.0202949Z 
2022-07-26T13:38:41.0203228Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/attributes.py:114: 
2022-07-26T13:38:41.0203713Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0203862Z 
2022-07-26T13:38:41.0203973Z args = (<EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0204345Z                 [(3.91923325e-13,     0.        ,...e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0204686Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>, (40,))
2022-07-26T13:38:41.0204948Z kwargs = {'subok': True}
2022-07-26T13:38:41.0205188Z relevant_args = (<EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0205574Z                 [(3.91923325e-13,     0.        ,...006592e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0205905Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>,)
2022-07-26T13:38:41.0206045Z 
2022-07-26T13:38:41.0206097Z >   ???
2022-07-26T13:38:41.0206195Z 
2022-07-26T13:38:41.0206295Z <__array_function__ internals>:180: 
2022-07-26T13:38:41.0206533Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0206671Z 
2022-07-26T13:38:41.0206792Z self = <EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0207226Z                 [(3.91923325e-13,     0.        , ...58006592e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0207556Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>
2022-07-26T13:38:41.0207811Z function = <function broadcast_to at 0x7ff05f782dd0>
2022-07-26T13:38:41.0208173Z types = (<class 'astropy.coordinates.earth.EarthLocation'>,)
2022-07-26T13:38:41.0208481Z args = (<EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0208839Z                 [(3.91923325e-13,     0.        ,...e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0209182Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>, (40,))
2022-07-26T13:38:41.0209424Z kwargs = {'subok': True}
2022-07-26T13:38:41.0209549Z 
2022-07-26T13:38:41.0209682Z     def __array_function__(self, function, types, args, kwargs):
2022-07-26T13:38:41.0209962Z         """Wrap numpy functions, taking care of units.
2022-07-26T13:38:41.0210162Z     
2022-07-26T13:38:41.0210328Z         Parameters
2022-07-26T13:38:41.0210545Z         ----------
2022-07-26T13:38:41.0210721Z         function : callable
2022-07-26T13:38:41.0210935Z             Numpy function to wrap
2022-07-26T13:38:41.0211157Z         types : iterable of classes
2022-07-26T13:38:41.0211407Z             Classes that provide an ``__array_function__`` override. Can
2022-07-26T13:38:41.0211707Z             in principle be used to interact with other classes. Below,
2022-07-26T13:38:41.0212010Z             mostly passed on to `~numpy.ndarray`, which can only interact
2022-07-26T13:38:41.0212264Z             with subclasses.
2022-07-26T13:38:41.0212443Z         args : tuple
2022-07-26T13:38:41.0212684Z             Positional arguments provided in the function call.
2022-07-26T13:38:41.0212985Z         kwargs : dict
2022-07-26T13:38:41.0213209Z             Keyword arguments provided in the function call.
2022-07-26T13:38:41.0213430Z     
2022-07-26T13:38:41.0213591Z         Returns
2022-07-26T13:38:41.0213785Z         -------
2022-07-26T13:38:41.0214024Z         result: `~astropy.units.Quantity`, `~numpy.ndarray`
2022-07-26T13:38:41.0214325Z             As appropriate for the function.  If the function is not
2022-07-26T13:38:41.0214648Z             supported, `NotImplemented` is returned, which will lead to
2022-07-26T13:38:41.0214954Z             a `TypeError` unless another argument overrode the function.
2022-07-26T13:38:41.0215285Z     
2022-07-26T13:38:41.0215442Z         Raises
2022-07-26T13:38:41.0215626Z         ------
2022-07-26T13:38:41.0215824Z         ~astropy.units.UnitsError
2022-07-26T13:38:41.0216160Z             If operands have incompatible units.
2022-07-26T13:38:41.0216343Z         """
2022-07-26T13:38:41.0216564Z         # A function should be in one of the following sets or dicts:
2022-07-26T13:38:41.0216844Z         # 1. SUBCLASS_SAFE_FUNCTIONS (set), if the numpy implementation
2022-07-26T13:38:41.0217119Z         #    supports Quantity; we pass on to ndarray.__array_function__.
2022-07-26T13:38:41.0217408Z         # 2. FUNCTION_HELPERS (dict), if the numpy implementation is usable
2022-07-26T13:38:41.0217698Z         #    after converting quantities to arrays with suitable units,
2022-07-26T13:38:41.0217958Z         #    and possibly setting units on the result.
2022-07-26T13:38:41.0218207Z         # 3. DISPATCHED_FUNCTIONS (dict), if the function makes sense but
2022-07-26T13:38:41.0218539Z         #    requires a Quantity-specific implementation.
2022-07-26T13:38:41.0218983Z         # 4. UNSUPPORTED_FUNCTIONS (set), if the function does not make sense.
2022-07-26T13:38:41.0219269Z         # For now, since we may not yet have complete coverage, if a
2022-07-26T13:38:41.0219554Z         # function is in none of the above, we simply call the numpy
2022-07-26T13:38:41.0219928Z         # implementation.
2022-07-26T13:38:41.0220141Z         if function in SUBCLASS_SAFE_FUNCTIONS:
2022-07-26T13:38:41.0220479Z >           return super().__array_function__(function, types, args, kwargs)
2022-07-26T13:38:41.0220649Z 
2022-07-26T13:38:41.0221072Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/units/quantity.py:1728: 
2022-07-26T13:38:41.0221374Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0221509Z 
2022-07-26T13:38:41.0221620Z array = <EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0221996Z                 [(3.91923325e-13,     0.        , ...58006592e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0222326Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>
2022-07-26T13:38:41.0222541Z shape = (40,), subok = True
2022-07-26T13:38:41.0222650Z 
2022-07-26T13:38:41.0222870Z     @array_function_dispatch(_broadcast_to_dispatcher, module='numpy')
2022-07-26T13:38:41.0223157Z     def broadcast_to(array, shape, subok=False):
2022-07-26T13:38:41.0223404Z         """Broadcast an array to a new shape.
2022-07-26T13:38:41.0223590Z     
2022-07-26T13:38:41.0223753Z         Parameters
2022-07-26T13:38:41.0223968Z         ----------
2022-07-26T13:38:41.0224141Z         array : array_like
2022-07-26T13:38:41.0224345Z             The array to broadcast.
2022-07-26T13:38:41.0224550Z         shape : tuple or int
2022-07-26T13:38:41.0224804Z             The shape of the desired array. A single integer ``i`` is interpreted
2022-07-26T13:38:41.0225051Z             as ``(i,)``.
2022-07-26T13:38:41.0225245Z         subok : bool, optional
2022-07-26T13:38:41.0225579Z             If True, then sub-classes will be passed-through, otherwise
2022-07-26T13:38:41.0225963Z             the returned array will be forced to be a base-class array (default).
2022-07-26T13:38:41.0226198Z     
2022-07-26T13:38:41.0226355Z         Returns
2022-07-26T13:38:41.0226543Z         -------
2022-07-26T13:38:41.0226780Z         broadcast : array
2022-07-26T13:38:41.0227039Z             A readonly view on the original array with the given shape. It is
2022-07-26T13:38:41.0227346Z             typically not contiguous. Furthermore, more than one element of a
2022-07-26T13:38:41.0227650Z             broadcasted array may refer to a single memory location.
2022-07-26T13:38:41.0227872Z     
2022-07-26T13:38:41.0228017Z         Raises
2022-07-26T13:38:41.0228222Z         ------
2022-07-26T13:38:41.0228400Z         ValueError
2022-07-26T13:38:41.0228688Z             If the array is not compatible with the new shape according to NumPy's
2022-07-26T13:38:41.0228950Z             broadcasting rules.
2022-07-26T13:38:41.0229130Z     
2022-07-26T13:38:41.0229277Z         See Also
2022-07-26T13:38:41.0229479Z         --------
2022-07-26T13:38:41.0229649Z         broadcast
2022-07-26T13:38:41.0229820Z         broadcast_arrays
2022-07-26T13:38:41.0230012Z         broadcast_shapes
2022-07-26T13:38:41.0230185Z     
2022-07-26T13:38:41.0230343Z         Notes
2022-07-26T13:38:41.0230525Z         -----
2022-07-26T13:38:41.0230747Z         .. versionadded:: 1.10.0
2022-07-26T13:38:41.0230931Z     
2022-07-26T13:38:41.0231076Z         Examples
2022-07-26T13:38:41.0231278Z         --------
2022-07-26T13:38:41.0231465Z         >>> x = np.array([1, 2, 3])
2022-07-26T13:38:41.0231664Z         >>> np.broadcast_to(x, (3, 3))
2022-07-26T13:38:41.0231866Z         array([[1, 2, 3],
2022-07-26T13:38:41.0232047Z                [1, 2, 3],
2022-07-26T13:38:41.0232214Z                [1, 2, 3]])
2022-07-26T13:38:41.0232387Z         """
2022-07-26T13:38:41.0232625Z >       return _broadcast_to(array, shape, subok=subok, readonly=True)
2022-07-26T13:38:41.0232794Z 
2022-07-26T13:38:41.0233142Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/numpy/lib/stride_tricks.py:413: 
2022-07-26T13:38:41.0233527Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0233651Z 
2022-07-26T13:38:41.0233766Z array = <EarthLocation [[(6.37813700e+03,     0.        ,     0.        )],
2022-07-26T13:38:41.0234113Z                 [(3.91923325e-13,     0.        , ...58006592e+01,  3196.81762195, -5500.8235441 )],
2022-07-26T13:38:41.0234483Z                 [(8.28240295e+02, -1434.55427221, -6140.69753401)]] km>
2022-07-26T13:38:41.0234704Z shape = (40,), subok = True, readonly = True
2022-07-26T13:38:41.0234837Z 
2022-07-26T13:38:41.0234953Z     def _broadcast_to(array, shape, subok, readonly):
2022-07-26T13:38:41.0235204Z         shape = tuple(shape) if np.iterable(shape) else (shape,)
2022-07-26T13:38:41.0235463Z         array = np.array(array, copy=False, subok=subok)
2022-07-26T13:38:41.0235695Z         if not shape and array.shape:
2022-07-26T13:38:41.0236026Z             raise ValueError('cannot broadcast a non-scalar to a scalar array')
2022-07-26T13:38:41.0236270Z         if any(size < 0 for size in shape):
2022-07-26T13:38:41.0370853Z             raise ValueError('all elements of broadcast shape must be non-'
2022-07-26T13:38:41.0371210Z                              'negative')
2022-07-26T13:38:41.0371388Z         extras = []
2022-07-26T13:38:41.0371587Z >       it = np.nditer(
2022-07-26T13:38:41.0371912Z             (array,), flags=['multi_index', 'refs_ok', 'zerosize_ok'] + extras,
2022-07-26T13:38:41.0372256Z             op_flags=['readonly'], itershape=shape, order='C')
2022-07-26T13:38:41.0372573Z E       ValueError: input operand has more dimensions than allowed by the axis remapping
2022-07-26T13:38:41.0372769Z 
2022-07-26T13:38:41.0373050Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/numpy/lib/stride_tricks.py:349: ValueError
2022-07-26T13:38:41.0373256Z 
2022-07-26T13:38:41.0373401Z During handling of the above exception, another exception occurred:
2022-07-26T13:38:41.0373561Z 
2022-07-26T13:38:41.0373651Z     def test_cirs_itrs_topo():
2022-07-26T13:38:41.0373838Z         """
2022-07-26T13:38:41.0374161Z         Check basic CIRS<->ITRS topocentric transforms for round-tripping.
2022-07-26T13:38:41.0374537Z         """
2022-07-26T13:38:41.0374843Z         locs = EarthLocation(lat=[[0], [90], [33.5], [-60], [-75]] * u.deg,
2022-07-26T13:38:41.0375179Z                              lon=[[0], [0], [45], [89], [-60]] * u.deg,
2022-07-26T13:38:41.0375483Z                              height=[[0], [1], [-0.1], [0.4], [2]] * u.km)
2022-07-26T13:38:41.0375715Z         usph = golden_spiral_grid(40)
2022-07-26T13:38:41.0376025Z >       cirs = CIRS(usph, obstime='J2000', location=locs)
2022-07-26T13:38:41.0376175Z 
2022-07-26T13:38:41.0376507Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/tests/test_intermediate_transformations.py:362: 
2022-07-26T13:38:41.0376832Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0377239Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/baseframe.py:313: in __init__
2022-07-26T13:38:41.0377539Z     values[fnm] = getattr(self, fnm)
2022-07-26T13:38:41.0377759Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-07-26T13:38:41.0377892Z 
2022-07-26T13:38:41.0378110Z self = <astropy.coordinates.attributes.EarthLocationAttribute object at 0x7ff05c885390>
2022-07-26T13:38:41.0378754Z instance = <[ValueError('attribute location should be scalar or have shape (40,), but is has shape (5, 1) and could not be broadcast.') raised in repr()] CIRS object at 0x7ff0502b52a0>
2022-07-26T13:38:41.0379408Z frame_cls = <class 'astropy.coordinates.builtin_frames.cirs.CIRS'>
2022-07-26T13:38:41.0379599Z 
2022-07-26T13:38:41.0379713Z     def __get__(self, instance, frame_cls=None):
2022-07-26T13:38:41.0379925Z         if instance is None:
2022-07-26T13:38:41.0380128Z             out = self.default
2022-07-26T13:38:41.0380314Z         else:
2022-07-26T13:38:41.0380602Z             out = getattr(instance, '_' + self.name, self.default)
2022-07-26T13:38:41.0380842Z             if out is None:
2022-07-26T13:38:41.0381110Z                 out = getattr(instance, self.secondary_attribute, self.default)
2022-07-26T13:38:41.0381336Z     
2022-07-26T13:38:41.0381543Z         out, converted = self.convert_input(out)
2022-07-26T13:38:41.0381847Z         if instance is not None:
2022-07-26T13:38:41.0382230Z             instance_shape = getattr(instance, 'shape', None)  # None if instance (frame) has no data!
2022-07-26T13:38:41.0382646Z             if instance_shape is not None and (getattr(out, 'shape', ()) and
2022-07-26T13:38:41.0382934Z                                                out.shape != instance_shape):
2022-07-26T13:38:41.0383206Z                 # If the shapes do not match, try broadcasting.
2022-07-26T13:38:41.0383417Z                 try:
2022-07-26T13:38:41.0383651Z                     if isinstance(out, ShapedLikeNDArray):
2022-07-26T13:38:41.0383947Z                         out = out._apply(np.broadcast_to, shape=instance_shape,
2022-07-26T13:38:41.0384198Z                                          subok=True)
2022-07-26T13:38:41.0384403Z                     else:
2022-07-26T13:38:41.0384653Z                         out = np.broadcast_to(out, instance_shape, subok=True)
2022-07-26T13:38:41.0384895Z                 except ValueError:
2022-07-26T13:38:41.0385133Z                     # raise more informative exception.
2022-07-26T13:38:41.0385365Z >                   raise ValueError(
2022-07-26T13:38:41.0385619Z                         "attribute {} should be scalar or have shape {}, "
2022-07-26T13:38:41.0385888Z                         "but is has shape {} and could not be broadcast."
2022-07-26T13:38:41.0386172Z                         .format(self.name, instance_shape, out.shape))
2022-07-26T13:38:41.0386533Z E                   ValueError: attribute location should be scalar or have shape (40,), but is has shape (5, 1) and could not be broadcast.
2022-07-26T13:38:41.0386760Z 
2022-07-26T13:38:41.0387044Z ../../.tox/py310-test-cov/lib/python3.10/site-packages/astropy/coordinates/attributes.py:117: ValueError
```

Reply:
Thanks! At least one of the failures is not quite what I expected, so that means implementing proper broadcasting may be a bit triicker; I linked to this discussion in #8812